bindings/guile/gnc-engine-guile.cpp

```diff
@@ -378,6 +380,8 @@ gnc_query_path2scm (const GSList *path)
     return scm_reverse (path_scm);
 }
 
+std::unordered_set<std::string> params;
```

Comment:
Needs to be `static` to keep it out of the global namespace.  But isn't this just a local reinvention of the string cache?

Reply:
Yes it's a string cache however I don't see where the strings refcount would be decremented.

Reply:
In [`gnc_query_path_free`](https://github.com/Gnucash/gnucash/blob/8f7eecd3e94df935549a7e8cecd155ba99ad122a/bindings/guile/gnc-engine-guile.cpp#L406). Which, BTW, will crash spectacularly with this change.

There's another use, [`%typemap(in) QofQueryParamList*`](https://github.com/Gnucash/gnucash/blob/8f7eecd3e94df935549a7e8cecd155ba99ad122a/bindings/engine.i#L224) that leaks the list and its contents. It needs a corresponding [freearg typemap](https://www.swig.org/Doc1.3/Typemaps.html#Typemaps_nn33) to clean up.

Reply:
Uh no because in general (non-bindings) code the `const char*` are static eg https://github.com/Gnucash/gnucash/blob/8f7eecd3e94df935549a7e8cecd155ba99ad122a/gnucash/import-export/csv-exp/assistant-csv-export.c#L630 and https://github.com/Gnucash/gnucash/blob/8f7eecd3e94df935549a7e8cecd155ba99ad122a/gnucash/gnome/gnc-split-reg.c#L1956 onwards

Reply:
Ok I didn't see that `gnc_query_path_free` is bindings only.

Reply:
Rewritten to use the built-in string cache.

Reply:
and swigged.

Reply:
...and swigging seems lotta trouble. reasoning: there are some static QofQueryParamList being used as described in https://github.com/Gnucash/gnucash/pull/1818#discussion_r1390017645 and some may be freed incorrectly.

Reply:
For example?

Reply:
Not sure... and I don't know how to run reports in builddir under gdb.

Reply:
You can run the report tests easily enough: Start with `ctest -V -R <test name>`. The top of the output is the command line and environment variables. Start with `gdb guile`, set up the environment (you might create a .gdb for that part, it gets tedious), set any breakpoints you want, and then `run -c <the Scheme fragment to start the test>`.

Tor run an actual report you might try `gdb bin/gnucash-cli` then set up the environment and `run -R run --name <report name> <test datafile>`.